connection.go

```diff
@@ -1650,6 +1650,22 @@ func (s *connection) handleCloseError(closeErr *closeError) {
 		s.tracer.ClosedConnection(e)
 	}
 
+	defer func() {
```

Comment:
Should this happen at the end of the run loop?

Reply:
That's where I had it originally, but moved it here to match the rest of cleanup code. I'll move it since I don't have a strong preference.

Reply:
Remember that this needs to happen after we remove ourselves from the packet handler. I've moved this to a defer in the run method.

Reply:
Feel free to move this somewhere else